apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ExtensionContext.java

```diff
@@ -100,11 +143,11 @@ public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass())
             return false;
         ExtensionContext that = (ExtensionContext) o;
-        return skipAnalysis == that.skipAnalysis;
+        return skipAnalysis == that.skipAnalysis && sendingTimestamp.equals(that.sendingTimestamp);
```

Comment:
`sendingTimestamp` may be `null`?

```suggestion
        return skipAnalysis == that.skipAnalysis && Objects.equals(sendingTimestamp, that.sendingTimestamp);
```